rs/boundary_node/ic_boundary/src/metrics.rs

```diff
@@ -651,18 +652,29 @@ pub async fn metrics_middleware(
             .observe(response_size as f64);
 
         // Anonymization
-        let s = anonymization_salt.load();
+        let salt = anonymization_salt.load();
 
-        let hash_fn = |v: &str| -> String {
-            if s.is_none() {
-                return "N/A".to_string();
-            }
+        let hash_fn = |input: &str| -> String {
+            let err_str = "N/A".to_string();
+
+            let salt = match salt.as_ref() {
+                Some(s) if s.len() == SALT_SIZE => s,
+                _ => return err_str,
+            };
+
+            // Create a 16-byte key for the SipHasher (uses half of the salt length)
+            let key: &[u8; 16] = match salt[0..16].try_into() {
```

Comment:
Since you already made sure that the salt is 32 bytes (above) you can just unwrap here.